src/core/datatypes/trees/TopologyNode.cpp

```diff
@@ -1762,6 +1764,163 @@ void TopologyNode::renameNodeParameter(const std::string &old_name, const std::s
+                {
+                    // get the age of the current child
+                    current_time = ages[i];
+                    
+                    // check if any extinct children become active
+                    size_t num_extinct = extinct_children.size();
+                    for (int j = num_extinct - 1; j >= 0; --j) // ditto
+                    {
+                        if ( extinct_children.at(j)->getAge() < current_time )
+                        {
+                            // add the extinct child to the active children list, remove it from the extinct children list
+                            active_children.push_back( extinct_children.at(j) );
+                            extinct_children.erase( extinct_children.begin() + long(j) );
+                        }
+                    }
+                    
+                    // randomly draw one child (arbitrarily called left) node from the list of active children
+                    size_t left = static_cast<size_t>( floor( rng->uniform01() * active_children.size() ) );
+                    TopologyNode* leftChild = active_children.at(left);
+                    
+                    // remove the randomly drawn node from the list
+                    active_children.erase( active_children.begin() + long(left) );
+                    
+                    // randomly draw one child (arbitrarily called right) node from the list of active children
+                    size_t right = static_cast<size_t>( floor( rng->uniform01() * active_children.size() ) );
+                    TopologyNode* rightChild = active_children.at(right);
+                    
+                    // remove the randomly drawn node from the list
+                    active_children.erase( active_children.begin() + long(right) );
+                    
+                    // remove the two also from the list of the children of the current node
+                    children.erase( std::remove(children.begin(), children.end(), leftChild), children.end() );
+                    children.erase( std::remove(children.begin(), children.end(), rightChild), children.end() );
+                    
+                    // create a parent for the two
+                    TopologyNode* prnt = new TopologyNode(); // leave the new node without index
+                    prnt->addChild( leftChild );
+                    prnt->addChild( rightChild );
+                    leftChild->setParent( prnt );
+                    rightChild->setParent( prnt );
+                    prnt->setAge( current_time );
+                    active_children.push_back( prnt );
+                    
+                    // add the newly created parent to the list of the children of the current node
+                    addChild( prnt );
+                    prnt->setParent( this );
+                }
+                
+            }
+            else
```

Comment:
I believe that you can have `use_ages = F` even if there is branch length information, but the function doesn't seem to account for it

Reply:
That's true: when the method was applied to a `BranchLengthTree`, the newly inserted branches had no branch lengths, which caused trouble when processing the resulting trees using `ape` (the missing branch lengths were represented as `NaN`).

I've now changed the code to give them a length of 0. This ensures that the original branch lengths and the original root-to-tip distances are both unchanged. The behavior for a `BranchLengthTree` is now exactly in line with how `ape::multi2di` works.

I also noticed it was possible for the branch subtending the multifurcating node to have its original length removed. This has been fixed as well.